### PR TITLE
fix(condo): DOMA-9808 use raw knex sum query in _allBillingReceiptsSum

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
A large load is expected on this endpoint, rewrote it more optimally.
Changed input type of query (it's internal api used only in `condo-restapi` service, it's not a breaking changes), leave only necessary fields in it, for ease of mapping to knex where

Before (~30k receipts returned by filter):
image

After:
image

### CODE REVIEW

File: apps/condo/domains/billing/schema/SumBillingReceiptsService.js [hunk 1]
```diff
@@ -4,72 +4,79 @@
...
-                const objs = await billingReceiptsLoader.load()
-                const toPaySum = objs.reduce((accumulator, receipt) => {
-                    return accumulator.plus(new Big(receipt.toPay))
-                }, new Big(0))
+                const { keystone } = getSchemaCtx('BillingReceipt')
+                const { knex } = getDatabaseAdapter(keystone)
+
+                const query = knex('BillingReceipt')
+                    .sum('BillingReceipt.toPay as totalToPay')
+                    .innerJoin('BillingIntegrationOrganizationContext', 'BillingReceipt.context', 'BillingIntegrationOrganizationContext.id')
```
Comment (toplenboren) [inline apps/condo/domains/billing/schema/SumBillingReceiptsService.js:52]: JOINS might be expensive too.. Maybe we can filter out required contexts using our business logic and then using these contexts just sum billing receipts?

Comment (toplenboren) [reply apps/condo/domains/billing/schema/SumBillingReceiptsService.js:52]: Might be worth a try, might be not

Comment (nomerdvadcatpyat) [reply apps/condo/domains/billing/schema/SumBillingReceiptsService.js]: Agree, remove additional joins from BillingReceipts query.
Change it to separate query for contexts and filter BillingReceipts by context ids

File: apps/condo/domains/billing/schema/SumBillingReceiptsService.js [hunk 2]
```diff
@@ -4,72 +4,91 @@
...
+                    organizationWhere.importRemoteSystem = importRemoteSystem
+                }
+
+                const contextIds = []
+                await allItemsQueryByChunks({
+                    schemaName: 'BillingIntegrationOrganizationContext',
+                    where: {
+                        organization: organizationWhere,
+                        deletedAt: null,
+                    },
```
Comment (Alllex202) [inline apps/condo/domains/billing/schema/SumBillingReceiptsService.js:68]: Should we take into account the integration status (`InProgress`/`Error`/`Finished`)? Only `Finished`?

Comment (toplenboren) [reply apps/condo/domains/billing/schema/SumBillingReceiptsService.js:68]: It does not matter in this case, It just should exist and be not deleted

**Review by toplenboren — APPROVED** [review]
Giving you a 👍 

Please - check submodule update (it should be pointing to main branch)
Please - check @Alllex202 comments

File: apps/condo/domains/billing/schema/SumBillingReceiptsService.js [hunk 3]
```diff
@@ -4,72 +4,91 @@
...
         },
     ],
-    
     queries: [
         {
             access: access.canSumBillingReceipts,
-            schema: '_allBillingReceiptsSum (where: BillingReceiptWhereInput!): BillingReceiptsSumOutput',
-            resolver: async (parent, args, context, info, extra = {}) => {
-                const { where } = args
+            schema: '_allBillingReceiptsSum (where: BillingReceiptsSumInput!): BillingReceiptsSumOutput',
```
[the remark this answers is not in the document]
Comment (nomerdvadcatpyat) [reply apps/condo/domains/billing/schema/SumBillingReceiptsService.js]: update parameter name from `where` to `data`

**Review by Alllex202 — APPROVED** [review]

**Comment by pahaz** [comment]
Do you have some speed tests for your changes?

File: apps/condo/domains/billing/schema/SumBillingReceiptsService.js [hunk 4]
```diff
@@ -4,72 +4,107 @@
...
+        message: 'Bad period format, must be YYYY-MM-01. Example: 2022-02-01',
     },
 }
 
 const SumBillingReceiptsService = new GQLCustomSchema('SumBillingReceiptsService', {
-
     types: [
+        {
+            access: true,
+            type: 'input BillingReceiptsSumInput { period: String!, organizationId: ID, tin: String, importRemoteSystem: String }',
```
Comment (pahaz) [inline apps/condo/domains/billing/schema/SumBillingReceiptsService.js:37]: `organizationId: ID` -- you need to use `organization: OrganizationWhereUniqInput`

Comment (pahaz) [reply apps/condo/domains/billing/schema/SumBillingReceiptsService.js:37]: Is it possible to keep only `organization` argument? The arg 'tin' seems redundant.

Comment (nomerdvadcatpyat) [reply apps/condo/domains/billing/schema/SumBillingReceiptsService.js:37]: > Is it possible to keep only `organization` argument? The arg 'tin' seems redundant.

No, search by tin in this query used in `condo-restapi` service

**Comment by nomerdvadcatpyat** [comment]
> Do you have some speed tests for your changes?

Read the pr description

**Pull request merged**